### PR TITLE
Refactor/details

### LINKED ISSUE #58: Front end error handling for bad request/server down

### PR DESCRIPTION
**Pull request by chrissycooper**
### Files Changed:
- `App.js` `apiCalls.js` `JournalEntry.js` and `NotFound.js`

### What Has Changed: 
- the `catch` block of our fetch calls was moved into `App` and `JournalEntry` so we could conditionally render an error message if an error was thrown. The content of the message depends on the status code number. 
- `setError` was passed into `JournalEntry` to be able to set the error state of App with its fetches. 
-  `NotFound` was adjusted to have a flexible error message.

### Notes: (includes questions, ideas for the future, anything you want to call out)
- I adjusted the end of the syntax of the `body` variable in `fetchCall` _mostly_ because I wanted to see if it would work and it did!! Please take that minor refactor as a suggestion. 
- There are some changes that I know @espressoGoddess  had generously taken on, I only added them so the linter would leave me alone. 
- More than happy to add anything else you think would make the error handling more robust tomorrow before we merge this PR in!

### Closes: (any issues that it closes)
closes #58

### CODE REVIEW

File: client/src/Components/App/App.js [hunk 1]
```diff
@@ -54,12 +57,16 @@ const App = () => {
 
   const updateLogs = () => {
     fetchCall()
-    .then(data => setUserLogs(data));
+    .then(data => setUserLogs(data))
+    .catch(err => {
+      setError(err);
+    });
```
Comment (jfogiato) [inline client/src/Components/App/App.js:63]: Love the simplicity in this! Kudos to @espressoGoddess, as well, for the insight into this!

File: client/src/Components/NotFound/NotFound.js [hunk 2]
```diff
@@ -1,16 +1,23 @@
 import React from 'react';
 import './NotFound.css';
+import PropTypes from 'prop-types';
 
-const NotFound = () => {
+const NotFound = ({err}) => {
+  const message = err?.message.includes('5') ? "There has been a server error." : "This page either moved or doesn't exist.";
```
Comment (jfogiato) [inline client/src/Components/NotFound/NotFound.js:6]: I'm super curious to know how to use a regex to check for a 500 level error but I also am exhausted

File: client/src/sampleData/activityData.js [hunk 3]
```diff
@@ -121,7 +121,7 @@ const activityData = [
   },
   {
     "activityId": 24,
-    "activity": "Call someone you haven't spoke with in a long time.",
+    "activity": "Call someone you haven't spoken with in a long time.",
```
Comment (jfogiato) [inline client/src/sampleData/activityData.js:124]: lololol grammar royalty 👑

File: client/src/utilities/api-calls.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 const fetchCall = (entry, id) => {
   const method = entry && id ? "PUT" : entry ? "POST" : "GET";
-  const body = entry?.date ? entry : entry ? {journal_entry: entry} : null;
+  const body = entry?.date ? entry : entry && {journal_entry: entry};
```
Comment (jfogiato) [inline client/src/utilities/api-calls.js:3]: YES! Love this!! The && operator consistently slips my mind, so thank you for snagging this!

**Review by jfogiato — APPROVED** [review]
Absolutely Grade A fantastic work on this CRC. This may feel slightly small and edge-case, but I think it will make our project look much more professional and thoughtful.

File: client/src/Components/App/App.js (hunk 1 above)
Comment (espressoGoddess) [reply client/src/Components/App/App.js:63]: yeeehaw! so clean and simple

File: client/src/Components/App/App.js [hunk 5]
```diff
@@ -54,12 +57,16 @@ const App = () => {
...
   }
 
   return (
     <main>
       <Header user={user}/>
+      {error ? <NotFound err={error}/> :
```
[the remark this answers is not in the document]
Comment (espressoGoddess) [reply client/src/Components/App/App.js:69]: 💯

File: client/src/Components/JournalEntry/JournalEntry.js [hunk 6]
```diff
@@ -72,9 +73,12 @@ JournalEntry.propTypes = {
...
     activity: PropTypes.string.isRequired,
     journal_entry: PropTypes.string.isRequired,
-    entry_ID: PropTypes.number,
+    entry_id: PropTypes.number,
     created_at: PropTypes.string,
     updated_at: PropTypes.string,
     user_id: PropTypes.string,
-  })
+  }),
+  updateLogs: PropTypes.func,
```
Comment (espressoGoddess) [inline client/src/Components/JournalEntry/JournalEntry.js:81]: nice thanks for adding this to proptypes!

File: client/src/sampleData/activityData.js (hunk 3 above)
Comment (espressoGoddess) [reply client/src/sampleData/activityData.js:124]: love the attention to detail

**Review by espressoGoddess — APPROVED** [review]

File: client/src/Components/App/App.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (KatherineBlaine) [reply client/src/Components/App/App.js:69]: SO smart and readable!

File: client/src/Components/NotFound/NotFound.js (hunk 2 above)
Comment (KatherineBlaine) [inline client/src/Components/NotFound/NotFound.js:6]: Love this error message!

File: client/src/Components/NotFound/NotFound.js [hunk 7]
```diff
@@ -1,16 +1,23 @@
...
+  
   return (
     <section className='notfound-container'>
-      <h1 className='notfound-header'>Oh no! This page either moved or doesn't exist. Try something else.</h1>
+      <h1 className='notfound-header'>Oh no! {message} Try something else.</h1>
```
Comment (KatherineBlaine) [inline client/src/Components/NotFound/NotFound.js:10]: Love the interpolation for specific error messaging!

File: client/src/sampleData/activityData.js (hunk 3 above)
Comment (KatherineBlaine) [inline client/src/sampleData/activityData.js:124]: What would we do without you, Chrissy?

**Review by KatherineBlaine — APPROVED** [review]
Thanks for updating our error handling! It is all so clean, concise, and thorough and adds a lot to our project!

**Pull request merged**